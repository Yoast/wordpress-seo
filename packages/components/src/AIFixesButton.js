import styled from "styled-components";
import { rgba } from "@yoast/style-guide";
import IconButtonBase from "./IconButtonBase";

const yoastPrimary50 = "#FAF3F7";
const yoastPrimary100 = "#F3E5ED";
const yoastPrimary300 = "#CD82AB";
const yoastPrimary400 = "#B94986";
const yoastPrimary500 = "#A61E69";
const yoastIndigo50 = "#EEF2FF";
const yoastIndigo100 = "#E0E7FF";
const yoastIndigo300 = "#A5B4FC";
const yoastIndigo500 = "#6366F1";

const direction = "to bottom right";

const gradientEffect = {
	defaultStateBackground: `linear-gradient(${direction}, ${yoastPrimary50}, ${yoastIndigo50})`,
	defaultStateBorder: `linear-gradient(${direction}, ${yoastPrimary300}, ${yoastIndigo300}) 1`,
	hoverStateBackground: `linear-gradient(${direction}, ${yoastPrimary100}, ${yoastIndigo100})`,
	pressedStateBackground: `linear-gradient(${direction}, ${yoastPrimary500}, ${yoastIndigo500})`,
};

const AIFixesButtonBase = styled( IconButtonBase )`
	position: relative;
	border: ${ props => props.pressed ? "none" : `1px solid ${yoastIndigo300}` };
	background-image: ${ props => props.pressed
		? gradientEffect.pressedStateBackground
		: gradientEffect.defaultStateBackground } !important;
	box-shadow: ${ props => props.pressed
		? `inset 0 -2px 0 ${ yoastPrimary400 }`
		: `0 1px 0 ${ rgba( props.unpressedBoxShadowColor, 0.7 ) }` };
	&:hover {
		background-image:  ${ props => props.pressed
			? gradientEffect.pressedStateBackground
			: gradientEffect.hoverStateBackground } !important;
	}
`;

export default AIFixesButtonBase;
