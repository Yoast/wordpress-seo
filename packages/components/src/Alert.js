/* External dependencies */
import PropTypes from "prop-types";
import React from "react";
import styled from "styled-components";
import { __ } from "@wordpress/i18n";

/* Yoast dependencies */
import { getDirectionalStyle } from "@yoast/helpers";
import { colors } from "@yoast/style-guide";

/* Internal dependencies */
import Button from "./buttons/Button";
import SvgIcon from "./SvgIcon";

const AlertContainer = styled.div`
	display: flex;
	align-items: flex-start;
	font-size: 14px;
	line-height: 1.5;
	border: 1px solid rgba(0, 0, 0, 0.2);
	padding: 16px;
	color: ${ props => props.alertColor };
	background: ${ props => props.alertBackground };
	margin-bottom: 20px;
`;

const AlertContent = styled.div`
	flex-grow: 1;

	a {
		color: ${ colors.$color_alert_link_text };
	}
`;

const AlertIcon = styled( SvgIcon )`
	margin-top: 0.125rem;
	${ getDirectionalStyle( "margin-right: 8px", "margin-left: 8px" ) };
`;

const AlertDismiss = styled( Button )`
	${ getDirectionalStyle( "margin: -8px -12px -8px 8px", "margin: -8px 8px -12px -8px" ) };
	font-size: 24px;
	line-height: 1.4;
	color: ${ props => props.alertDismissColor };
	flex-shrink: 0;
	min-width: 36px;
	height: 36px;

	// Override the base button style: get rid of the button styling.
	padding: 0;

	&, &:hover, &:active {
		/* Inherits box-sizing: border-box so this doesn't change the rendered size. */
		border: 2px solid transparent;
		background: transparent;
		box-shadow: none;
		color: ${ props => props.alertDismissColor };
	}

	/* Inherits focus style from the Button component. */
	&:focus {
		background: transparent;
		color: ${ props => props.alertDismissColor };
		border-color: ${ colors.$color_yoast_focus };
		box-shadow: 0px 0px 0px 3px ${ colors.$color_yoast_focus_outer };
	}
`;

/**
 * Alert component.
 *
 * @param {Object}   props               The props to use.
 * @param {*}        props.children      The children to render inside the alert.
 * @param {string}   props.type          The type of Alert. Can be: "error", "info", "success" or "warning".
 *                                       Controls the colors and icon of the Alert.
 * @param {Function} [props.onDismissed] When supplied this Alert will gain an 'X' button.
 *                                       Note: the function provided should do the actual dismissing!
 * @param {string}   dismissAriaLabel    The close button aria-label. Must be a translatable string with
 *                                       text domain.
 *
 * @returns {Alert} The Alert component.
 */
class Alert extends React.Component {
	/**
	 * Returns the colors and icon to be used based on the type provided to the props.
	 *
	 * @param {string} type The type of Alert.
	 *
	 * @returns {object} Options with colors and icons to be used.
	 */
	getTypeDisplayOptions( type ) {
		switch ( type ) {
			case "error":
				return {
					color: colors.$color_alert_error_text,
					background: colors.$color_alert_error_background,
					icon: "alert-error",
				};
			case "info":
				return {
					color: colors.$color_alert_info_text,
					background: colors.$color_alert_info_background,
					icon: "alert-info",
				};
			case "success":
				return {
					color: colors.$color_alert_success_text,
					background: colors.$color_alert_success_background,
					icon: "alert-success",
				};
			case "warning":
				return {
					color: colors.$color_alert_warning_text,
					background: colors.$color_alert_warning_background,
					icon: "alert-warning",
				};
		}
	}

	/**
	 * Renders the component.
	 *
	 * @returns {React.Element} The rendered component.
	 */
	render() {
		if ( this.props.isAlertDismissed === true ) {
			return null;
		}

		const options          = this.getTypeDisplayOptions( this.props.type );
		const dismissAriaLabel = this.props.dismissAriaLabel || __( "Dismiss this alert", "yoast-components" );

		return <AlertContainer alertColor={ options.color } alertBackground={ options.background } className={ this.props.className }>
			<AlertIcon icon={ options.icon } color={ options.color } />
			<AlertContent>{ this.props.children }</AlertContent>
			{
				typeof this.props.onDismissed === "function"
					? (
						<AlertDismiss
							alertDismissColor={ options.color }
							onClick={ this.props.onDismissed }
							aria-label={ dismissAriaLabel }
						>
							&times;
						</AlertDismiss>
					)
					: null
			}
		</AlertContainer>;
	}
}

Alert.propTypes = {
	children: PropTypes.any.isRequired,
	type: PropTypes.oneOf( [ "error", "info", "success", "warning" ] ).isRequired,
	onDismissed: PropTypes.func,
	isAlertDismissed: PropTypes.bool,
	dismissAriaLabel: PropTypes.string,
	className: PropTypes.string,
};

Alert.defaultProps = {
	onDismissed: null,
	isAlertDismissed: false,
	dismissAriaLabel: "",
	className: "",
};

export default Alert;
