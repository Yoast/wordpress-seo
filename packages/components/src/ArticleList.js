/* External dependencies */
import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

/* Yoast dependencies */
import { makeOutboundLink } from "@yoast/helpers";

/**
 * @typedef  {Object}     Feed
 * @property {string}     title       The title of the website.
 * @property {string}     description A description of the website.
 * @property {string}     link        A link to the website.
 * @property {FeedItem[]} items       The items in the feed.
 */

/**
 * @typedef  {Object} FeedItem
 * @property {string} title       The title of the item.
 * @property {string} content     The content of the item, will be HTML encoded.
 * @property {string} description A summary of the content, will be HTML encoded.
 * @property {string} link        A link to the item.
 * @property {string} creator     The creator of the item.
 * @property {string} date        The publication date of the item.
 */

const ArticleListContainer = styled.div`
	box-sizing: border-box;

	p {
		margin: 0;
		font-size: 14px;
	}
`;

const ArticleListHeader = styled.h3`
	margin: 8px 0;
	font-size: 1em;
`;

const FeedList = styled.ul`
	margin: 0;
	list-style: none;
	padding: 0;
`;

const ArticleListLink = makeOutboundLink( styled.a`
	display: inline-block;
	margin-bottom: 4px;
	font-size: 14px;
` );

const ArticleListItemContainer = styled.li`
	margin: 8px 0;
`;

const ArticleListFooter = styled.div`
	a {
		margin: 8px 0 0;
	}
`;

/**
 * The WordpressFeedListItem component.
 *
 * @param {object} props The component's props.
 *
 * @returns {ReactElement} The WordpressFeedListItem component.
 */
const ArticleListItem = ( props ) => {
	return (
		<ArticleListItemContainer
			className={ props.className }
		>
			<ArticleListLink
				className={ `${ props.className }-link` }
				href={ props.link }
			>
				{ props.title }
			</ArticleListLink>
			<p className={ `${ props.className }-description` }>
				{ props.description }
			</p>
		</ArticleListItemContainer>
	);
};

ArticleListItem.propTypes = {
	className: PropTypes.string.isRequired,
	title: PropTypes.string.isRequired,
	link: PropTypes.string.isRequired,
	description: PropTypes.string.isRequired,
};

/**
 * Displays a parsed wordpress feed.
 *
 * @param {Object} props                The component props.
 * @param {Feed}   props.feed           The feed object.
 * @param {string} props.title          The title. Defaults to feed title.
 * @param {string} props.footerLinkText The footer link text.
 * @param {string} props.feedLink       The footer link. Defaults to feed link.
 *
 * @returns {ReactElement} The ArticleList component.
 */
const ArticleList = ( props ) => {
	return (
		<ArticleListContainer
			className={ props.className }
		>
			<ArticleListHeader
				className={ `${ props.className }__header` }
			>
				{ props.title ? props.title : props.feed.title }
			</ArticleListHeader>
			<FeedList
				className={ `${ props.className }__posts` }
				role="list"
			>
				{ props.feed.items.map( item => (
					<ArticleListItem
						className={ `${ props.className }__post` }
						key={ item.link }
						title={ item.title }
						link={ item.link }
						description={ item.description }
					/>
				) ) }
			</FeedList>
			{ props.footerLinkText &&
			<ArticleListFooter
				className={ `${ props.className }__footer` }
			>
				<ArticleListLink
					className={ `${ props.className }__footer-link` }
					href={ props.feedLink ? props.feedLink : props.feed.link }
				>
					{ props.footerLinkText }
				</ArticleListLink>
			</ArticleListFooter>
			}
		</ArticleListContainer>
	);
};

ArticleList.propTypes = {
	className: PropTypes.string,
	feed: PropTypes.object.isRequired,
	title: PropTypes.string,
	footerLinkText: PropTypes.string,
	feedLink: PropTypes.string,
};

ArticleList.defaultProps = {
	className: "articlelist-feed",
};

export default ArticleList;
