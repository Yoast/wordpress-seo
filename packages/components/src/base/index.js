import "./borders.css";
import "./colors.css";
import "./icons.css";
import "./typography.css";
import "./utility.css";
