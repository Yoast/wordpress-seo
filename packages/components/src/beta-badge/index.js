import "../badges/badge.css";
import "./beta-badge.css";
export { default as BetaBadge } from "./BetaBadge";
