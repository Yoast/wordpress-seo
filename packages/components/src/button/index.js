import "./buttons.css";

export { Button as NewButton, ButtonStyledLink } from "./Button.js";
export { CloseButton } from "./Icon.js";
