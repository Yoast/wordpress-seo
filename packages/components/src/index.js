import "./base";
import "./title-separator";

import StyledSection, { StyledHeading, StyledSectionBase } from "./StyledSection";
import { LinkButton } from "./buttons/LinkButton";

export * from "./button";
export * from "./checkbox";
export * from "./data-model";
export * from "./field-group";
export * from "./image-select";
export * from "./inputs";
export * from "./insights-card";
export * from "./radiobutton";
export * from "./select";
export * from "./star-rating";
export * from "./help-icon";
export * from "./tables";
export * from "./new-badge";
export * from "./premium-badge";

// Referenced index.js explicitly due to case-sensitive path conflicts.
export * from "./toggle/index.js";

export {
	StyledSection,
	StyledSectionBase,
	StyledHeading,
	LinkButton,
};

export {
	default as Button,
	BaseButton,
	addHoverStyle,
	addActiveStyle,
	addFocusStyle,
	addBaseStyle,
	addButtonStyles,
} from "./buttons/Button";

export {
	default as Collapsible,
	CollapsibleStateless,
	StyledIconsButton,
	StyledContainer,
	StyledContainerTopLevel,
	wrapInHeading,
} from "./Collapsible";

export { default as Alert } from "./Alert";
export { default as ArticleList } from "./ArticleList";
export { default as Card, FullHeightCard } from "./Card";
export { default as CardBanner } from "./CardBanner";
export { default as CourseDetails } from "./CourseDetails";
export { default as IconLabeledButton } from "./buttons/IconLabeledButton";
export { default as IconButton } from "./buttons/IconButton";
export { default as IconsButton } from "./buttons/IconsButton";
export { default as ErrorBoundary } from "./ErrorBoundary";
export { default as Heading } from "./Heading";
export { default as HelpText } from "./HelpText";
export { default as Icon } from "./Icon";
export { default as IconButtonToggle } from "./IconButtonToggle.js";
export { default as IFrame } from "./IFrame";
export { default as Input } from "./input/Input";
export { default as WordOccurrenceInsights } from "./WordOccurrenceInsights";
export { default as Label, SimulatedLabel } from "./Label";
export { default as LanguageNotice, languageNoticePropType } from "./LanguageNotice";
export { default as Loader } from "./Loader";
export { default as MultiStepProgress } from "./MultiStepProgress";
export { default as Notification } from "./Notification";
export { default as Paper } from "./Paper";
export { default as ProgressBar } from "./ProgressBar";
export { default as Section } from "./Section";
export { SectionTitle } from "./SectionTitle";
export { default as ScoreAssessments } from "./ScoreAssessments";
export { default as StackedProgressBar } from "./StackedProgressBar";
export { default as SvgIcon, icons } from "./SvgIcon";
export { default as SynonymsInput } from "./SynonymsInput";
export { default as Textarea } from "./Textarea";
export { default as Textfield } from "./Textfield";
export { default as Toggle } from "./Toggle";
export { UpsellButton } from "./buttons/UpsellButton";
export { UpsellLinkButton } from "./buttons/UpsellLinkButton";
export { YoastButton } from "./buttons/YoastButton";
export { InputField } from "./input/InputField";
export { YoastLinkButton } from "./buttons/YoastLinkButton";
export { default as Logo } from "./Logo";
export { default as Modal } from "./Modal";
export { default as YoastSeoIcon } from "./YoastSeoIcon";
export { default as Tabs } from "./Tabs";
export { default as Warning } from "./Warning";
export { default as YouTubeVideo } from "./YouTubeVideo";
export { default as WordList } from "./WordList";
export { default as WordOccurrences } from "./WordOccurrences";
export { VariableEditorInputContainer } from "./input/InputContainer";
export { default as InsightsCard } from "./insights-card/InsightsCard";

export { ListTable, ZebrafiedListTable } from "./table/ListTable";
export { Row } from "./table/Row";

export { default as ScreenReaderText } from "./a11y/ScreenReaderText";
export { default as ScreenReaderShortcut } from "./a11y/ScreenReaderShortcut";

// Deprecated:
export { default as KeywordSuggestions } from "./WordOccurrenceInsights";
