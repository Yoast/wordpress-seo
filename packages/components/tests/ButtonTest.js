// External dependencies.
import React from "react";
import renderer from "react-test-renderer";

// Internal dependencies.
import { IconButton, BaseButton, Button } from "../src/index";

test( "the BaseButton matches the snapshot", () => {
	const component = renderer.create(
		<BaseButton>ButtonValue</BaseButton>
	);

	const tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );

test( "BaseButton executes callback", () => {
	const component = renderer.create(
		<BaseButton
			onClick={
				() => {
					return "clicked";
				}
			}
		>ButtonValue</BaseButton>
	);

	let tree = component.toJSON();
	expect( tree ).toMatchSnapshot();

	tree.props.onClick();

	tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );

test( "the SnippetPreviewButton matches the snapshot", () => {
	const component = renderer.create(
		<Button>ButtonValue</Button>
	);

	const tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );

test( "SnippetPreviewButton executes callback", () => {
	const component = renderer.create(
		<Button
			onClick={
				() => {
					return "clicked";
				}
			}
		>ButtonValue</Button>
	);

	let tree = component.toJSON();
	expect( tree ).toMatchSnapshot();

	tree.props.onClick();

	tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );

test( "the IconButton matches the snapshot", () => {
	const component = renderer.create(
		<IconButton icon="edit" iconColor="black" />
	);

	const tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );

test( "the IconButton with text matches the snapshot", () => {
	const component = renderer.create(
		<IconButton icon="edit" iconColor="black">Click</IconButton>
	);

	const tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );
