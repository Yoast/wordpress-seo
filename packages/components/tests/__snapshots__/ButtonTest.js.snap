// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseButton executes callback 1`] = `
.c0 {
  color: #555;
  border-color: #ccc;
  background: #f7f7f7;
  box-shadow: 0 1px 0 rgba( 204,204,204,1 );
}

.c1:active {
  color: #000;
  background-color: #f7f7f7;
  box-shadow: inset 0 2px 5px -3px rgba( 0,0,0,0.5 );
}

.c2::-moz-focus-inner {
  border-width: 0;
}

.c2:focus {
  outline: none;
  border-color: #0066cd;
  color: #000;
  background-color: #fff;
  box-shadow: 0 0 3px rgba( 8,74,103,0.8 );
}

.c3:hover {
  color: #000;
  background-color: #fff;
  border-color: var(--yoast-color-border--default);
}

.c4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  vertical-align: middle;
  border-width: 1px;
  border-style: solid;
  margin: 0;
  padding: 4px 10px;
  border-radius: 3px;
  cursor: pointer;
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  font-weight: inherit;
  text-align: left;
  overflow: visible;
  min-height: 32px;
  -webkit-transition: var(--yoast-transition-default);
  transition: var(--yoast-transition-default);
}

.c4 svg {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .c4::after {
    display: inline-block;
    content: "";
    min-height: 22px;
  }
}

<button
  className="c0 c1 c2 c3 c4"
  onClick={[Function]}
  type="button"
>
  ButtonValue
</button>
`;

exports[`BaseButton executes callback 2`] = `
.c0 {
  color: #555;
  border-color: #ccc;
  background: #f7f7f7;
  box-shadow: 0 1px 0 rgba( 204,204,204,1 );
}

.c1:active {
  color: #000;
  background-color: #f7f7f7;
  box-shadow: inset 0 2px 5px -3px rgba( 0,0,0,0.5 );
}

.c2::-moz-focus-inner {
  border-width: 0;
}

.c2:focus {
  outline: none;
  border-color: #0066cd;
  color: #000;
  background-color: #fff;
  box-shadow: 0 0 3px rgba( 8,74,103,0.8 );
}

.c3:hover {
  color: #000;
  background-color: #fff;
  border-color: var(--yoast-color-border--default);
}

.c4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  vertical-align: middle;
  border-width: 1px;
  border-style: solid;
  margin: 0;
  padding: 4px 10px;
  border-radius: 3px;
  cursor: pointer;
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  font-weight: inherit;
  text-align: left;
  overflow: visible;
  min-height: 32px;
  -webkit-transition: var(--yoast-transition-default);
  transition: var(--yoast-transition-default);
}

.c4 svg {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .c4::after {
    display: inline-block;
    content: "";
    min-height: 22px;
  }
}

<button
  className="c0 c1 c2 c3 c4"
  onClick={[Function]}
  type="button"
>
  ButtonValue
</button>
`;

exports[`SnippetPreviewButton executes callback 1`] = `
.c0 {
  color: #555;
  border-color: #ccc;
  background: #f7f7f7;
  box-shadow: 0 1px 0 rgba( 204,204,204,1 );
}

.c1:active {
  color: #000;
  background-color: #f7f7f7;
  box-shadow: inset 0 2px 5px -3px rgba( 0,0,0,0.5 );
}

.c2::-moz-focus-inner {
  border-width: 0;
}

.c2:focus {
  outline: none;
  border-color: #0066cd;
  color: #000;
  background-color: #fff;
  box-shadow: 0 0 3px rgba( 8,74,103,0.8 );
}

.c3:hover {
  color: #000;
  background-color: #fff;
  border-color: var(--yoast-color-border--default);
}

.c4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  vertical-align: middle;
  border-width: 1px;
  border-style: solid;
  margin: 0;
  padding: 4px 10px;
  border-radius: 3px;
  cursor: pointer;
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  font-weight: inherit;
  text-align: left;
  overflow: visible;
  min-height: 32px;
  -webkit-transition: var(--yoast-transition-default);
  transition: var(--yoast-transition-default);
}

.c4 svg {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .c4::after {
    display: inline-block;
    content: "";
    min-height: 22px;
  }
}

<button
  className="c0 c1 c2 c3 c4"
  onClick={[Function]}
  type="button"
>
  ButtonValue
</button>
`;

exports[`SnippetPreviewButton executes callback 2`] = `
.c0 {
  color: #555;
  border-color: #ccc;
  background: #f7f7f7;
  box-shadow: 0 1px 0 rgba( 204,204,204,1 );
}

.c1:active {
  color: #000;
  background-color: #f7f7f7;
  box-shadow: inset 0 2px 5px -3px rgba( 0,0,0,0.5 );
}

.c2::-moz-focus-inner {
  border-width: 0;
}

.c2:focus {
  outline: none;
  border-color: #0066cd;
  color: #000;
  background-color: #fff;
  box-shadow: 0 0 3px rgba( 8,74,103,0.8 );
}

.c3:hover {
  color: #000;
  background-color: #fff;
  border-color: var(--yoast-color-border--default);
}

.c4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  vertical-align: middle;
  border-width: 1px;
  border-style: solid;
  margin: 0;
  padding: 4px 10px;
  border-radius: 3px;
  cursor: pointer;
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  font-weight: inherit;
  text-align: left;
  overflow: visible;
  min-height: 32px;
  -webkit-transition: var(--yoast-transition-default);
  transition: var(--yoast-transition-default);
}

.c4 svg {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .c4::after {
    display: inline-block;
    content: "";
    min-height: 22px;
  }
}

<button
  className="c0 c1 c2 c3 c4"
  onClick={[Function]}
  type="button"
>
  ButtonValue
</button>
`;

exports[`the BaseButton matches the snapshot 1`] = `
.c0 {
  color: #555;
  border-color: #ccc;
  background: #f7f7f7;
  box-shadow: 0 1px 0 rgba( 204,204,204,1 );
}

.c1:active {
  color: #000;
  background-color: #f7f7f7;
  box-shadow: inset 0 2px 5px -3px rgba( 0,0,0,0.5 );
}

.c2::-moz-focus-inner {
  border-width: 0;
}

.c2:focus {
  outline: none;
  border-color: #0066cd;
  color: #000;
  background-color: #fff;
  box-shadow: 0 0 3px rgba( 8,74,103,0.8 );
}

.c3:hover {
  color: #000;
  background-color: #fff;
  border-color: var(--yoast-color-border--default);
}

.c4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  vertical-align: middle;
  border-width: 1px;
  border-style: solid;
  margin: 0;
  padding: 4px 10px;
  border-radius: 3px;
  cursor: pointer;
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  font-weight: inherit;
  text-align: left;
  overflow: visible;
  min-height: 32px;
  -webkit-transition: var(--yoast-transition-default);
  transition: var(--yoast-transition-default);
}

.c4 svg {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .c4::after {
    display: inline-block;
    content: "";
    min-height: 22px;
  }
}

<button
  className="c0 c1 c2 c3 c4"
  type="button"
>
  ButtonValue
</button>
`;

exports[`the IconButton matches the snapshot 1`] = `
.c5 {
  width: 16px;
  height: 16px;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
}

.c0 {
  color: #555;
  border-color: #ccc;
  background: #f7f7f7;
  box-shadow: 0 1px 0 rgba( 204,204,204,1 );
}

.c1:active {
  color: #000;
  background-color: #f7f7f7;
  box-shadow: inset 0 2px 5px -3px rgba( 0,0,0,0.5 );
}

.c2::-moz-focus-inner {
  border-width: 0;
}

.c2:focus {
  outline: none;
  border-color: #0066cd;
  color: #000;
  background-color: #fff;
  box-shadow: 0 0 3px rgba( 8,74,103,0.8 );
}

.c3:hover {
  color: #000;
  background-color: #fff;
  border-color: var(--yoast-color-border--default);
}

.c4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  vertical-align: middle;
  border-width: 1px;
  border-style: solid;
  margin: 0;
  padding: 4px 10px;
  border-radius: 3px;
  cursor: pointer;
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  font-weight: inherit;
  text-align: left;
  overflow: visible;
  min-height: 32px;
  -webkit-transition: var(--yoast-transition-default);
  transition: var(--yoast-transition-default);
}

.c4 svg {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .c4::after {
    display: inline-block;
    content: "";
    min-height: 22px;
  }
}

<button
  className="c0 c1 c2 c3 c4"
  type="button"
>
  <svg
    aria-hidden={true}
    className="c5 yoast-svg-icon yoast-svg-icon-edit"
    fill="black"
    focusable="false"
    role="img"
    size="16px"
    viewBox="0 0 1792 1792"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M491 1536l91-91-235-235-91 91v107h128v128h107zm523-928q0-22-22-22-10 0-17 7l-542 542q-7 7-7 17 0 22 22 22 10 0 17-7l542-542q7-7 7-17zm-54-192l416 416-832 832h-416v-416zm683 96q0 53-37 90l-166 166-416-416 166-165q36-38 90-38 53 0 91 38l235 234q37 39 37 91z"
    />
  </svg>
</button>
`;

exports[`the IconButton with text matches the snapshot 1`] = `
.c5 {
  width: 16px;
  height: 16px;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
}

.c0 {
  color: #555;
  border-color: #ccc;
  background: #f7f7f7;
  box-shadow: 0 1px 0 rgba( 204,204,204,1 );
}

.c1:active {
  color: #000;
  background-color: #f7f7f7;
  box-shadow: inset 0 2px 5px -3px rgba( 0,0,0,0.5 );
}

.c2::-moz-focus-inner {
  border-width: 0;
}

.c2:focus {
  outline: none;
  border-color: #0066cd;
  color: #000;
  background-color: #fff;
  box-shadow: 0 0 3px rgba( 8,74,103,0.8 );
}

.c3:hover {
  color: #000;
  background-color: #fff;
  border-color: var(--yoast-color-border--default);
}

.c4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  vertical-align: middle;
  border-width: 1px;
  border-style: solid;
  margin: 0;
  padding: 4px 10px;
  border-radius: 3px;
  cursor: pointer;
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  font-weight: inherit;
  text-align: left;
  overflow: visible;
  min-height: 32px;
  -webkit-transition: var(--yoast-transition-default);
  transition: var(--yoast-transition-default);
}

.c4 svg {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c6 {
  margin: 0 8px 0 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .c4::after {
    display: inline-block;
    content: "";
    min-height: 22px;
  }
}

<button
  className="c0 c1 c2 c3 c4"
  type="button"
>
  <svg
    aria-hidden={true}
    className="c5 yoast-svg-icon yoast-svg-icon-edit c6"
    fill="black"
    focusable="false"
    role="img"
    size="16px"
    viewBox="0 0 1792 1792"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M491 1536l91-91-235-235-91 91v107h128v128h107zm523-928q0-22-22-22-10 0-17 7l-542 542q-7 7-7 17 0 22 22 22 10 0 17-7l542-542q7-7 7-17zm-54-192l416 416-832 832h-416v-416zm683 96q0 53-37 90l-166 166-416-416 166-165q36-38 90-38 53 0 91 38l235 234q37 39 37 91z"
    />
  </svg>
  Click
</button>
`;

exports[`the SnippetPreviewButton matches the snapshot 1`] = `
.c0 {
  color: #555;
  border-color: #ccc;
  background: #f7f7f7;
  box-shadow: 0 1px 0 rgba( 204,204,204,1 );
}

.c1:active {
  color: #000;
  background-color: #f7f7f7;
  box-shadow: inset 0 2px 5px -3px rgba( 0,0,0,0.5 );
}

.c2::-moz-focus-inner {
  border-width: 0;
}

.c2:focus {
  outline: none;
  border-color: #0066cd;
  color: #000;
  background-color: #fff;
  box-shadow: 0 0 3px rgba( 8,74,103,0.8 );
}

.c3:hover {
  color: #000;
  background-color: #fff;
  border-color: var(--yoast-color-border--default);
}

.c4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  vertical-align: middle;
  border-width: 1px;
  border-style: solid;
  margin: 0;
  padding: 4px 10px;
  border-radius: 3px;
  cursor: pointer;
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  font-weight: inherit;
  text-align: left;
  overflow: visible;
  min-height: 32px;
  -webkit-transition: var(--yoast-transition-default);
  transition: var(--yoast-transition-default);
}

.c4 svg {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .c4::after {
    display: inline-block;
    content: "";
    min-height: 22px;
  }
}

<button
  className="c0 c1 c2 c3 c4"
  type="button"
>
  ButtonValue
</button>
`;
